Add option in memory planning to put shared state on same location across entry points

JacobSzwejbka opened the pull request:
Summary: API that lets you place the same state tensor on the same id and offset across entry points. Lets you have get and set state more natively in the runtime if the underlying arenas are the same.

Differential Revision: D82250153

GregoryComer reviewed (approved):
I don't have full context on the memory planning logic, but this looks reasonable to me. This will be a very nice feature for encoder-decoder models. As an aside, do you think it will be feasible to default share_mutable_buffers to true in the future?

JacobSzwejbka committed:
Add option in memory planning to put shared state on same location across entry points (#14230)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/14230

API that lets you place the same state tensor on the same id and offset across entry points. Lets you have get and set state more natively in the runtime if the underlying arenas are the same.

Reviewed By: GregoryComer

Differential Revision: D82250153
